cli/src/lib.rs

```diff
@@ -3378,7 +3490,97 @@ fn stream_logs(config: &WithPath<Config>, rpc_url: &str) -> Result<Vec<std::proc
     Ok(handles)
 }
 
-fn start_test_validator(
+fn start_surfpool_validator(
+    flags: Option<Vec<String>>,
+    surfpool_config: &Option<SurfpoolConfig>,
+) -> Result<Child> {
+    let rpc_url = surfpool_rpc_url(surfpool_config);
+    let mut validator_handle = std::process::Command::new("surfpool")
+        .arg("start")
+        .args(flags.unwrap_or_default())
+        .spawn()
+        .map_err(|e| anyhow!("Failed to spawn `surfpool`: {e}"))?;
+
+    let client = create_client(rpc_url.clone());
+
+    let mut count = 0;
+
+    let ms_wait = surfpool_config
+        .as_ref()
+        .map(|surfpool| surfpool.startup_wait)
+        .unwrap_or(STARTUP_WAIT);
+
+    while count < ms_wait {
+        let r = client.get_latest_blockhash();
+        if r.is_ok() {
+            break;
+        }
+        std::thread::sleep(std::time::Duration::from_millis(100));
+        count += 100;
+    }
+
+    let mut runbook_completed = false;
+
+    while !runbook_completed {
+        let rpc_request = serde_json::json!({
+            "jsonrpc": "2.0",
+            "id": 1,
+            "method": "surfnet_getSurfnetInfo"
+        });
+
+        let http_client = reqwest::blocking::Client::new();
+
+        match http_client
+            .post(&rpc_url)
+            .header("Content-Type", "application/json")
+            .json(&rpc_request)
+            .send()
+        {
+            Ok(response) => {
+                if response.status().is_success() {
+                    match response.json::<serde_json::Value>() {
```

Comment:
If you made a little struct to represent the runbook executions, you could deserialize straight to that struct. The approach you have is fine I think, more of personal taste. But I did want to bring up the option.